sycl/source/detail/queue_impl.cpp

```diff
@@ -278,6 +278,12 @@ void queue_impl::wait(const detail::code_location &CodeLoc) {
   TelemetryEvent = instrumentationProlog(CodeLoc, Name, StreamID, IId);
 #endif
 
+  if (has_property<ext::oneapi::property::queue::lazy_execution>()) {
+    const detail::plugin &Plugin = getPlugin();
+    if (Plugin.getBackend() == backend::ext_oneapi_level_zero)
+      Plugin.call<detail::PiApiKind::piQueueFlush>(getHandleRef());
+  }
```

Comment:
What happens if I call wait on an event instead? Will it have flushed somewhere else or will I stall?

Also, do we need to check for L0 here? It would probably be better to stop the user from even passing the property to any other backend, as I suspect if/once another backend adds support for lazy execution this is easy to miss and will be hard to find.